FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Oliver Jordy Pérez Escamilla on 23/01/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public final class CoreDataFeedStore: FeedStore {
+	
+	// MARK: Properties
+	lazy var context: NSManagedObjectContext = {
+		Self.coreDataStack.context
+	}()
+	private static var coreDataStack: CoreDataStack = CoreDataStack(modelName: "FeedStoreModel")
+	
+	// MARK: Initializers
+	public init(coreDataStack: CoreDataStack) {
+		Self.coreDataStack = coreDataStack
+	}
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		context.perform { [unowned context] in
+			
+			do {
+				guard let cache = try self.fetchCache() else { return completion(nil) }
+				context.delete(cache)
+				
+				completion(nil)
```

Comment:
It should only complete after saving the operation:

```suggestion
				context.delete(cache)
				try context.save()
				completion(nil)
```